### ISSUE TITLE
Setup connection pool by default for Rails cache

### ISSUE THREAD

**Issue by eoinkelly** [ackama/rails-template#199]
When setting up Rails cache with Redis as the cache store it creates on connection per puma *process* but this is sub-optimal because each puma process has `RAILS_MAX_THREADS` threads (usually 5) so this can lead to the threads having to wait for a connection. 

The [rails caching guide](https://edgeguides.rubyonrails.org/caching_with_rails.html#connection-pool-options) explains how to do it and there is more info in https://github.com/rails/rails/issues/39479

**Comment by eoinkelly** [comment]
This is an example of how we did this on previous client projects:

```ruby
# config/environments/production.rb
# ...
    config.cache_store = :redis_cache_store, {
      url: ENV.fetch("RAILS_CACHE_REDIS_URL"),

      ##
      # Enable connection pooling because we use it in production and want to
      # exercise the same code paths in development.
      #
      # Configuring a connection pool for Redis as Rails cache is documented in:
      #
      # * https://edgeguides.rubyonrails.org/caching_with_rails.html#connection-pool-options
      #
      # but some more details are available in:
      #
      # * https://github.com/rails/rails/blob/a5d1628c79ab89dfae57ec1e1aeca467e29de188/activesupport/lib/active_support/cache.rb#L168-L173
      # * https://github.com/rails/rails/blob/9b4aef4be3dc58eb08f694387857b52be8050954/activesupport/lib/active_support/cache/redis_cache_store.rb#L185-L192
      #
      pool_size: Integer(ENV.fetch("RAILS_MAX_THREADS", 5)), # number of connections **per process**
      pool_timeout: 5 # num seconds to wait for a connection
    }

# ...

**Comment by eoinkelly** [comment]
Closed by #296
